DSharpPlus.Interactivity/EventHandling/ComponentBased/Requests/ButtonPaginationRequest.cs

```diff
@@ -0,0 +1,183 @@
+// This file is part of the DSharpPlus project.
+//
+// Copyright (c) 2015 Mike Santiago
+// Copyright (c) 2016-2021 DSharpPlus Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DSharpPlus.Entities;
+using DSharpPlus.Interactivity.Enums;
+
+namespace DSharpPlus.Interactivity.EventHandling
+{
+    internal class ButtonPaginationRequest : IPaginationRequest
+    {
+        private int _index;
+        private readonly List<Page> _pages = new();
+
+        private readonly TaskCompletionSource<bool> _tcs = new();
+
+        private readonly CancellationToken _token;
+        private readonly DiscordUser _user;
+        private readonly DiscordMessage _message;
+        private readonly PaginationButtons _buttons;
+        private readonly PaginationBehaviour _wrapBehavior;
+        private readonly ButtonPaginationBehavior _behaviorBehavior;
+
+        public ButtonPaginationRequest(DiscordMessage message, DiscordUser user,
+            PaginationBehaviour behavior, ButtonPaginationBehavior behaviorBehavior,
+            PaginationButtons buttons, Page[] pages, CancellationToken token)
```

Comment:
`IEnumerable<Page>` instead of `Page[]` maybe?

Reply:
Sure thing